hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1086,39 +1086,44 @@ private void initialCommit(String createInstantTime, List<MetadataPartitionType>
 
     Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionToRecordsMap = new HashMap<>();
 
-    List<DirectoryInfo> partitionInfoList = listAllPartitions(dataMetaClient);
-    Map<String, Map<String, Long>> partitionToFilesMap = partitionInfoList.stream()
-        .map(p -> {
-          String partitionName = HoodieTableMetadataUtil.getPartitionIdentifier(p.getRelativePath());
-          return Pair.of(partitionName, p.getFileNameToSizeMap());
-        })
-        .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
-
-    int totalDataFilesCount = partitionToFilesMap.values().stream().mapToInt(Map::size).sum();
-    List<String> partitions = new ArrayList<>(partitionToFilesMap.keySet());
-
-    if (partitionTypes.contains(MetadataPartitionType.FILES)) {
-      // Record which saves the list of all partitions
-      HoodieRecord allPartitionRecord = HoodieMetadataPayload.createPartitionListRecord(partitions);
-      HoodieData<HoodieRecord> filesPartitionRecords = getFilesPartitionRecords(createInstantTime, partitionInfoList, allPartitionRecord);
-      ValidationUtils.checkState(filesPartitionRecords.count() == (partitions.size() + 1));
-      partitionToRecordsMap.put(MetadataPartitionType.FILES, filesPartitionRecords);
-    }
+    // skip parsing file system for metadata records if its first commit for the table.
+    if (dataMetaClient.getActiveTimeline().getWriteTimeline().countInstants() > 1
+        || dataMetaClient.getActiveTimeline().getWriteTimeline().filterCompletedInstants().countInstants() != 0) {
+      List<DirectoryInfo> partitionInfoList = listAllPartitions(dataMetaClient);
```

Comment:
> dataMetaClient.getActiveTimeline().getWriteTimeline().countInstants() > 1

How about we have 2 inflight instants on the timeline?

Reply:
+1 is there a cleaner way of checking this? this feels a bit obtuse to derive that this is the first commit into the MDT.

Reply:
I have changed the way we deduce the first commit. essentially piggy backing on the initial commit instant time. If its SOLO_COMMIT_TIMESTAMP("00000000000000"), then we know its a fresh table. If not, its an already existing table.